public/general_components/multi_select_filter.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFilterSelectItem,
+  EuiFilterGroup,
+  EuiPopover,
+  EuiFilterButton,
+  EuiFlexItem,
+} from '@elastic/eui';
+
+// styling
+import './general-component-styles.scss';
+
+interface MultiSelectFilterProps {
+  title: string;
+  filters: EuiFilterSelectItem[];
+  setSelectedFilters: (filters: EuiFilterSelectItem[]) => void;
```

Comment:
We have this setter as a prop since typically any filter changes will need to be propagated to the parent component(s).